### PR TITLE
Feature/ajax fetch

### PR DESCRIPTION
**Pull request by UnderTheKnife**

### CODE REVIEW

File: AjaxFetch/js/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        function(data){
+          localStorage.setItem(path, data);
+        });
+    });
+}
+
+
+window.addEventListener('load',() => {
+
+  window.values = {
```
Comment (StasVysotskyi) [inline AjaxFetch/js/index.js]: You can use this variable without window. Just define that on the top of document

File: AjaxFetch/js/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    users: JSON.parse(localStorage.getItem('users')),
+    posts: JSON.parse(localStorage.getItem('posts')),
+    comments: JSON.parse(localStorage.getItem('comments')),
+    albums: JSON.parse(localStorage.getItem('albums')),
+    photos: JSON.parse(localStorage.getItem('photos'))
+  };
+
+  $.each(window.values, (index, value) => {
+    if (value === null) {
```
Comment (StasVysotskyi) [inline AjaxFetch/js/index.js]: You can use short form, like `if (!value)`

File: AjaxFetch/js/show.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
+/**
+ * show users
```
Comment (StasVysotskyi) [inline AjaxFetch/js/show.js]: Start all of your JSDoc with a capital letter

File: AjaxFetch/js/show.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+ * @param {string} parent
+ */
+function user(parent) {
+  $.each(window.values.users, (userId, user) => {
+    let posts = $(`<div style="display: none"></div>`);
```
Comment (StasVysotskyi) [inline AjaxFetch/js/show.js]: Don't use inline styles. For hide this element, you can use **.hide() ** method of jQuery

File: AjaxFetch/js/show.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    let albums = $(`<div style="display: none"></div>`);
+    let userName = $(`<h2 class="well">${user.name}</h2>`);
```
Comment (StasVysotskyi) [inline AjaxFetch/js/show.js]: Try to use this method of creating elements with jQuery: 
```let userName = $(`<h2/>').addClass('well').text(user.name);```
This is better way for understanding your code.

File: AjaxFetch/js/show.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+ * show user albums
+ * @param {string} parent
+ * @param {number} parentId
+ */
+function album(parent, parentId) {
+  let allAlbums = $('<h3>Albums</h3>');
+  let albums = $('<div style="display: none"></div>');
+
+  $.each(window.values.albums, (albumId, album) => {
+    if (album['userId'] === parentId) {
```
Comment (StasVysotskyi) [inline AjaxFetch/js/show.js]: album, in this scope has type of Object. So, I think you can call userId of title like:
```album. userId ```
or not?

**Comment by UnderTheKnife** [comment]
refactored

File: AjaxFetch/css/styles.css [hunk 7]
```diff
@@ -0,0 +1,19 @@
+.well{
```
Comment (StasVysotskyi) [inline AjaxFetch/css/styles.css]: Check your css style in all files. Should be
```
.well {

```
Try to setup css lint for this problem

File: AjaxFetch/css/styles.css [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  min-height: 20px;
+  padding: 19px;
+  margin: 0 auto 20px;
+  background-color: #f5f5f5;
+  border: 1px solid #e3e3e3;
+  border-radius: 4px;
+  -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,.05);
```
Comment (StasVysotskyi) [inline AjaxFetch/css/styles.css]: You also can setup https://www.npmjs.com/package/gulp-autoprefixer, and don't write this param in your style.

File: AjaxFetch/css/styles.css [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  box-shadow: inset 0 1px 1px rgba(0,0,0,.05);
```
Comment (StasVysotskyi) [inline AjaxFetch/css/styles.css]:
```
rgba(0, 0, 0, .5);
```
Is it true?

File: AjaxFetch/js/index.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  const root = 'https://jsonplaceholder.typicode.com';
+
+  fetch(`${root}/${path}`).then(
+    function (response) {
+      response.json().then(
+        function (data) {
+          localStorage.setItem(path, JSON.stringify(data));
+        });
+    });
+}
```
Comment (StasVysotskyi) [inline AjaxFetch/js/index.js]: You can separate your callback method into new function and call it like:

```
fetch(`${root}/${path}`)
  .then(response => response.json())
  .then(data => yourCallBackMethod.bind(this, data))
  .catch(error => yourCallBackMethodForFailed.bind(this, error));
```

And decelerate your method line:

```
function yourCallBackMethod(data) {}
```

And one more things, please add catch for all of your Promises. You need to catch errors.
Please read:
https://learn.javascript.ru/promise
https://habrahabr.ru/post/199456/

File: AjaxFetch/js/show.js [hunk 11]
```diff
@@ -0,0 +1,113 @@
+/**
+ * Show users
+ * @param {string} parent
+ */
+function user(parent) {
+  $.each(values.users, (userId, user) => {
+    let posts = $(`<div/>`).hide();
+    let albums = $(`<div/>`).hide();
+    let userName = $('<h2/>').addClass('well').text(user.name);
```
Comment (StasVysotskyi) [inline AjaxFetch/js/show.js]: All of those variable can be const?

File: AjaxFetch/js/show.js [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+/**
+ * Show user posts
+ * @param {string} parent
+ * @param {number} parentId
+ */
+function post(parent, parentId) {
+  let allPosts = $('<h3/>').text('Posts');
+  let posts = $('<div/>').hide();
```
Comment (StasVysotskyi) [inline AjaxFetch/js/show.js]: All of those variable can be const?

**Review by StasVysotskyi — APPROVED** [review]

**Pull request merged**